JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeRoute.kt

```diff
@@ -106,7 +112,13 @@ fun HomeRoute(
         }
     }
 
-    val homeScreenType = getHomeScreenType(showTopAppBar, uiState)
+    val homeScreenType = getHomeScreenType(windowSize, uiState)
+
+    val showTopAppBar = when (windowSize) {
```

Comment:
I moved this decision to here: Only this route cares about whether to show the top app bar, or switch it out.